src/public/js/mainpage.js

```diff
@@ -17,15 +19,22 @@ $('#submitFileBtn').click((e) => {
       return;
     }
     updateHiddenFormContents('fileUpload', fileContentHash);
-    document.forms.fileUpload.submit();
+    if (recaptchaEnabled) {
+      grecaptcha.execute();
+    } else {
+      submitForm();
+    }
   };
   const file = $('#uploadFile').prop('files')[0];
   reader.readAsBinaryString(file);
-});
+  return false;
+};
 
-$(document).ready(() => {
+$(() => {
+  $('#fileUpload').on('submit', doFileUpload);
+  window.submitForm = submitForm;
   const fileInput = $('#uploadFileDiv input[type=file]');
-  fileInput.change(() => {
+  fileInput.on('change', () => {
```

Comment:
My ide kept complaining about the obsolete syntax, but given #8 this will soon become irrelevant.